src/old-tests/util.js

```diff
@@ -0,0 +1,239 @@
+}
+
+// ==================================================================
+
+export async function getNetworkId(xo) {
+  // const config = await loadConfig('').then(config);
+  const networks = xo.objects.indexes.type.network;
+  const network = find(networks, { name_label: config.network });
+  return network.id;
+}
+
+// ==================================================================
+
+export async function getVmXoTestPvId(xo) {
+  // const config = await loadConfig('').then(config);
+  const vms = xo.objects.indexes.type.VM;
+  const vm = find(vms, { name_label: config.pvVm });
+  return vm.id;
+}
+
+export async function getVmToMigrateId(xo) {
+  // config = await loadConfig('').then(config);
+  const vms = xo.objects.indexes.type.VM;
+  const vm = find(vms, { name_label: config.vmToMigrate });
+  return vm.id;
+}
+
+// ==================================================================
+
+export async function getSrId(xo) {
+  const host = getOneHost(xo);
+  const pool = await xo.getOrWaitObject(host.$poolId);
+  return pool.default_SR;
+}
+
+// ==================================================================
+
+export async function jobTest(xo) {
+  const vmId = await getVmXoTestPvId(xo);
+  const jobId = await xo.call("job.create", {
+    job: {
+      type: "call",
+      key: "snapshot",
+      method: "vm.snapshot",
+      paramsVector: {
+        type: "cross product",
+        items: [
+          {
+            type: "set",
+            values: [
+              {
+                id: vmId,
+                name: "snapshot",
+              },
+            ],
+          },
+        ],
+      },
+    },
+  });
+  return jobId;
+}
+
+export async function scheduleTest(xo, jobId) {
+  const schedule = await xo.call("schedule.create", {
+    jobId: jobId,
+    cron: "* * * * * *",
+    enabled: false,
+  });
+  return schedule;
+}
+
+export async function getSchedule(xo, id) {
+  const schedule = xo.call("schedule.get", { id: id });
+  return schedule;
+}
+
+// ==================================================================
+
+export function deepDelete(obj, path) {
+  const lastIndex = path.length - 1;
+  for (let i = 0; i < lastIndex; i++) {
+    obj = obj[path[i]];
+
+    if (typeof obj !== "object" || obj === null) {
+      return;
+    }
+  }
+  delete obj[path[lastIndex]];
+}
+
+export function almostEqual(actual, expected, ignoredAttributes) {
+  actual = cloneDeep(actual);
+  expected = cloneDeep(expected);
+  forEach(ignoredAttributes, ignoredAttribute => {
+    deepDelete(actual, ignoredAttribute.split("."));
+    deepDelete(expected, ignoredAttribute.split("."));
+  });
+  expect(actual).to.be.eql(expected);
+}
+
+export async function waitObjectState(xo, id, predicate) {
+  let obj = xo.objects.all[id];
+  while (true) {
+    try {
+      await predicate(obj);
+      return;
+    } catch (_) {}
+    // If failed, wait for next object state/update and retry.
+    obj = await xo.waitObject(id);
+  }
+}
```

Comment:
Why this new file?

Reply:
To keep a copy of the old `util.js`. 
Do I have to delete it?

Reply:
Yes, we are using Git to keep the history of files, we don't need to keep copies.